components/Typography/src/UIFont+MaterialScalable.h

```diff
@@ -46,6 +46,24 @@ typedef NSDictionary<UIContentSizeCategory, NSNumber *> *MDCScalingCurve;
  */
 - (nonnull UIFont *)mdc_scaledFontForSizeCategory:(nonnull UIContentSizeCategory)sizeCategory;
 
+/**
+ Returns a font with the same family, weight and traits, but whose point size is based on the given
+ trait environment's preferred content size category.
+
+ If the device is running iOS 9 and not in an extension, then the provided traitEnvironment will be
+ ignored and the UIApplication sharedApplication's preferredContentSizeCategory will be used
+ instead.
+
+ If the device is running iOS 9 and in an extension, then the provided trait environment will be
+ ignored and the returned font will be scaled with UIContentSizeCategoryLarge.
+
+ @param traitEnvironment The trait environment whose trait collection should be queried.
+ @return A font whose point size is extracted from @c mdc_scalingCurve for the given trait
```

Comment:
Nit: s/extracted/determined/ (or derived)? I believe the scaling is a ratio of the `.Large` point size based on the ratio of the `.Large` scaling factor and the current ContentSizeCategory scaling factor.